src/main/java/core/jdbc/JdbcTemplate.java

```diff
@@ -1,52 +1,58 @@
 package core.jdbc;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.PreparedStatementSetter;
 
-public class JdbcTemplate {
+import java.util.List;
 
-    public static void insert(String query, Object... values) throws SQLException {
+public class JdbcTemplate extends AbstractJdbcTemplate{
+
+    public void insert(String query, Object... values) throws DataAccessException {
         modify(query, values);
     }
 
-    public static void update(String query, Object... values) throws SQLException {
+    public void insert(String query, PreparedStatementSetter setter) throws DataAccessException {
+        modify(query, setter);
+    }
+
+    public void update(String query, Object... values) throws DataAccessException {
         modify(query, values);
     }
 
-    private static void modify(String query, Object... values) throws SQLException{
-        init(query, values).executeUpdate();
+    public void update(String query, PreparedStatementSetter setter) throws DataAccessException {
+        modify(query, setter);
     }
 
-    public static ResultSet select(String query, Object... values) throws SQLException{
-        return init(query, values).executeQuery();
+    public List<?> queryForList(String query, RowMapper<?> rowMapper, Object... values) throws DataAccessException{
+        return connectionAndSelect(query, rowMapper, values);
     }
 
-    private static PreparedStatement init(String query, Object... values) throws SQLException{
-        Connection con = null;
-        PreparedStatement pstmt = null;
-        con = ConnectionManager.getConnection();
-        pstmt = con.prepareStatement(query);
-        initValue(pstmt, values);
-        return pstmt;
+    public List<?> queryForList(String query, RowMapper<?> rowMapper, PreparedStatementSetter setter) throws DataAccessException{
```

Comment:
JdbcTemplate 사용처에서 아래와 같이 타입 캐스팅이 발생하네요.
https://github.com/next-step/jwp-jdbc/blob/a3afa18fbc262dab9d5dec1b77ca911f7c67076d/src/main/java/next/dao/UserDao.java#L52-L54


와일드카드 대신 타입 매개변수로 사용 해주세요.
```suggestion
    public <T> List<T> queryForList(String query, RowMapper<T> rowMapper, PreparedStatementSetter setter) throws DataAccessException{
```

제네릭에 대한 아티클들이 많이 있는데, 그 중 잘 정리되어 보이는 글을 첨부 드리니 참고 부탁드려요.
[[Java] 제네릭, 와일드카드 ](https://movefast.tistory.com/74)